LSPS2/README.md

```diff
@@ -0,0 +1,864 @@
+
+`lsps2.getversions` has a result like the below:
+
+```JSON
+{
+  "versions": [1]
+}
+```
+
+`versions` is the set of LSPS2 versions the LSP supports.
+When the client later contacts the LSP, it indicates a single specific
+version, which MUST be one of those indicated in this set.
+
+The client MUST determine protocol compatibility (if it supports a
+`version` that the LSP also supports).
+
+### 1. API Information
+
+`lsps2.getinfo` is the entrypoint for each client using the API.
+It indicates supported versions of this protocol, as well as any limits the
+LSP imposes, and parameters for payment.
+
+The client MUST request `lsps2.getinfo` to read the `opening_fee` of the
+LSP and its related parameters.
+
+`lsps2.getinfo` takes the parameters:
+
+```JSON
+{
+  "version": 1,
+  "token": "SECRETDISCOUNTCOUPON100"
+}
+```
+
+`version` is the version of this spec that will be used for this
+interaction.
+
+`token` is an *optional*, arbitrary JSON string.
+This parameter is intended for use between the client and the LSP; it
+may be used to negotiate for a better feerate than the LSP otherwise
+offers (i.e. a "discount coupon"), or for the LSP to actually offer
+this service to the client (i.e. an "API key") instead of failing to
+provide any offers, or for any other purpose.
+
+`lsps2.getinfo` has the following errors defined (error code numbers
+in parantheses):
+
+* `unsupported_version` (1) - the LSP does not support the `version`
+  indicated by the client.
+* `unrecognized_or_stale_token` (2) - the client provided the `token`,
+  and the LSP does not recognize it, or the token has expired.
+
+Example `lsps2.getinfo` result:
+
+```JSON
+{
+    "opening_fee_params_menu": [
+        {
+            "base_msat": "546000",
+            "proportional": 1200,
+            "valid_until": "2023-02-23T08:47:30.511Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        },
+        {
+            "base_msat": "1092000",
+            "proportional": 2400,
+            "valid_until": "2023-02-27T21:23:57.984Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        }
+    ],
+    "min_payment_size_msat": "1000",
+    "max_payment_size_msat": "1000000",
+}
+```
+The `opening_fee_params_menu` is an array of `opening_fee_params`
+objects.
+The LSP MAY return an empty array; in which case, the client currently
+cannot use JIT Channels with this LSP.
+
+An `opening_fee_params` object describes how much the LSP will charge
+for a channel open, and until when the described feerates will be
+considered valid.
+An `opening_fee_params` object MUST have all of the following fields,
+and MUST NOT have any additional fields:
+
+* `base_msat` is the number of millisatoshis that the client has to pay for the
+  channel open, regardless of how large the payment is.
```

Comment:
Because the proportional part of `opening_fee` here is computed off `payment_size` (general hum requested to do that instead of channel size), and `payment_size` is in millisatoshi and out of our control (e.g. it could even be variable-amount invoice, so payer might decide to pay 111,111,111,111 miillisats because they can).

I guess we could round up the proportional part but *shrug*.  Just go millisats here.